bubblegum/xrf/model/guessparam.py

```diff
@@ -43,15 +43,42 @@
 import logging
 logger = logging.getLogger(__name__)
 
-from atom.api import Atom, Str, observe, Typed, Int, Dict, List
+from pprint import pprint
+from atom.api import Atom, Str, observe, Typed, Int, Dict, List, Float, Enum
 
-from skxray.fitting.xrf_model import (ModelSpectrum, set_range, k_line, l_line, m_line,
-                                      get_linear_model, PreFitAnalysis)
+from skxray.fitting.xrf_model import (ModelSpectrum, set_range, k_line, l_line,
+                                      m_line, get_linear_model, PreFitAnalysis)
 from skxray.fitting.background import snip_method
 
 
-# This is not a right way to define path. To be updated
-data_path = '/Users/Li/Research/X-ray/Research_work/all_code/nsls2_gui/nsls2_gui/xrf_parameter.json'
+class Parameter(Atom):
+    # todo make sure that these are the only valid bound types
+    bound_type = Enum('none', 'lohi', 'fixed')
+    min = Float(-np.inf)
+    max = Float(np.inf)
+    value = Float()
+    default_value = Float()
+    free_more = Enum('none', 'fixed', 'lohi')
+    adjust_element = Enum('none', 'fixed', 'lohi')
+    e_calibration = Enum('none', 'fixed', 'lohi')
+    linear = Enum('none', 'fixed')
+    name = Str()
+    description = Str()
+    tool_tip = Str()
+
+    @observe('name')
+    def update_displayed_name(self, changed):
+        if not self.description:
+            self.description = self.name
+
+    def __repr__(self):
+        return ("Parameter(bound_type={}, min={}, max={}, value={}, "
+               "default_value={}, free_more={}, adjust_element={}, "
+               "e_calibration={}, linear={}, name={}, description={}, "
+               "toop_tip={}".format(
+            self.bound_type, self.min, self.max, self.value, self.default_value,
+            self.free_more, self.adjust_element, self.e_calibration,
+            self.linear, self.name, self.description, self.tool_tip))
```

Comment:
Got it. So this change will not penetrate to the fitting engine, but only on the GUI level, right? The whole back-end fitting is coupled with the original dictionary format.